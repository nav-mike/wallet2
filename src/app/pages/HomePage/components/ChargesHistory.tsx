import React from 'react';
import { VictoryArea, VictoryChart } from 'victory';

type Charge = {
  x: string;
  y: number;
};

type HistoryProps = {
  charges: Charge[];
};

const style = {
  data: {
    fill: 'navy',
    fillOpacity: 0.7,
    stroke: '#c43a31',
    strokeWidth: 3,
  },
  labels: {
    fill: 'white',
  },
  axis: {
    fill: 'white',
  },
};

export function ChargesHistory(props: HistoryProps) {
  return (
    <VictoryChart>
      <VictoryArea
        data={props.charges}
        style={style}
        labels={({ datum }) => datum.y}
      />
    </VictoryChart>
  );
}
