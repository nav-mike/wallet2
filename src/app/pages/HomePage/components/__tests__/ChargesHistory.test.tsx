import * as React from 'react';
import { render } from '@testing-library/react';
import { ChargesHistory } from '../ChargesHistory';

describe('<ChargesHistory />', () => {
  const data = [
    { x: '01.01.21', y: 23 },
    { x: '02.01.21', y: 33.2 },
  ];

  it('should render and match the snapshot', () => {
    const history = render(<ChargesHistory charges={data} />);
    expect(history.container.firstChild).toMatchSnapshot();
  });
});
