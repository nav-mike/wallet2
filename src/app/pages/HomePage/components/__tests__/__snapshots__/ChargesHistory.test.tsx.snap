// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChargesHistory /> should render and match the snapshot 1`] = `
<div
  class="VictoryContainer"
  style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
>
  <svg
    height="300"
    role="img"
    style="pointer-events: all; width: 100%; height: 100%;"
    viewBox="0 0 450 300"
    width="450"
  >
    <g
      clip-path="url(#victory-clip-2)"
      style="height: 100%; width: 100%; user-select: none;"
    >
      <defs>
        <clippath
          id="victory-clip-2"
        >
          <rect
            height="200"
            vector-effect="non-scaling-stroke"
            width="350"
            x="50"
            y="50"
          />
        </clippath>
      </defs>
      <g>
        <path
          d="M50,111.44578313253012L400,50L400,250L50,250Z"
          role="presentation"
          shape-rendering="auto"
          style="fill: navy; fill-opacity: 0.7; stroke: none; stroke-width: 3;"
        />
        <path
          d="M50,111.44578313253012L400,50"
          role="presentation"
          shape-rendering="auto"
          style="fill: none; fill-opacity: 0.7; stroke: #c43a31; stroke-width: 3;"
        />
      </g>
    </g>
    <g
      role="presentation"
    >
      <line
        role="presentation"
        shape-rendering="auto"
        style="stroke: #252525; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
        vector-effect="non-scaling-stroke"
        x1="50"
        x2="400"
        y1="250"
        y2="250"
      />
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-1-tickLabels-0"
          x="50"
          y="272.97"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="middle"
            x="50"
          >
            01.01.21
          </tspan>
        </text>
      </g>
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-1-tickLabels-1"
          x="400"
          y="272.97"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="middle"
            x="400"
          >
            02.01.21
          </tspan>
        </text>
      </g>
    </g>
    <g
      role="presentation"
    >
      <line
        role="presentation"
        shape-rendering="auto"
        style="stroke: #252525; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
        vector-effect="non-scaling-stroke"
        x1="50"
        x2="50"
        y1="50"
        y2="250"
      />
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-2-tickLabels-0"
          x="39"
          y="224.84951807228916"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="end"
            x="39"
          >
            5
          </tspan>
        </text>
      </g>
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-2-tickLabels-1"
          x="39"
          y="194.72903614457832"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="end"
            x="39"
          >
            10
          </tspan>
        </text>
      </g>
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-2-tickLabels-2"
          x="39"
          y="164.60855421686747"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="end"
            x="39"
          >
            15
          </tspan>
        </text>
      </g>
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-2-tickLabels-3"
          x="39"
          y="134.48807228915663"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="end"
            x="39"
          >
            20
          </tspan>
        </text>
      </g>
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-2-tickLabels-4"
          x="39"
          y="104.36759036144579"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="end"
            x="39"
          >
            25
          </tspan>
        </text>
      </g>
      <g
        role="presentation"
      >
        <text
          direction="inherit"
          dx="0"
          id="chart-axis-2-tickLabels-5"
          x="39"
          y="74.24710843373495"
        >
          <tspan
            dx="0"
            dy="0"
            style="font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; fill: #252525; stroke: transparent;"
            text-anchor="end"
            x="39"
          >
            30
          </tspan>
        </text>
      </g>
    </g>
  </svg>
  <div
    style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
  >
    <svg
      height="300"
      style="overflow: visible; width: 100%; height: 100%;"
      viewBox="0 0 450 300"
      width="450"
    >
      <text
        direction="inherit"
        dx="0"
        id="chart-area-0-labels-0"
        x="50"
        y="99.41578313253012"
      >
        <tspan
          dx="0"
          dy="0"
          style="fill: white; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
          text-anchor="middle"
          x="50"
        >
          23
        </tspan>
      </text>
      <text
        direction="inherit"
        dx="0"
        id="chart-area-0-labels-1"
        x="400"
        y="37.97"
      >
        <tspan
          dx="0"
          dy="0"
          style="fill: white; font-family: 'Gill Sans', 'Seravek', 'Trebuchet MS', sans-serif; font-size: 14px; letter-spacing: normal; padding: 10px; stroke: transparent;"
          text-anchor="middle"
          x="400"
        >
          33.2
        </tspan>
      </text>
    </svg>
  </div>
</div>
`;
